### ISSUE TITLE
TypeError: object of type 'map' has no len()

### ISSUE THREAD

**Issue by monajalal** [cleardusk/EmotionChallenge#1]
Using Python2.7 I get this error:

```
(py2emotion) [jalal@goku crop_align]$ /scratch/sjn-p2/anaconda/anaconda2/bin/python landmark.py
Traceback (most recent call last):
  File "landmark.py", line 7, in <module>
    import dlib
  File "/scratch/sjn-p2/anaconda/anaconda2/lib/python2.7/site-packages/dlib/__init__.py", line 1, in <module>
    from .dlib import *
ImportError: libcudnn.so.6: cannot open shared object file: No such file or directory
(py2emotion) [jalal@goku crop_align]$ which conda
/scratch/anaconda3/envs/py2emotion/bin/conda
(py2emotion) [jalal@goku crop_align]$ source deactivate py2emotion
[jalal@goku crop_align]$ /scratch/sjn-p2/anaconda/anaconda2/bin/python landmark.py
Traceback (most recent call last):
  File "landmark.py", line 7, in <module>
    import dlib
  File "/scratch/sjn-p2/anaconda/anaconda2/lib/python2.7/site-packages/dlib/__init__.py", line 1, in <module>
    from .dlib import *
ImportError: libcudnn.so.6: cannot open shared object file: No such file or directory
[jalal@goku crop_align]$ /scratch/sjn-p2/anaconda/anaconda2/bin/conda install cudnn=6.0
Fetching package metadata .............
Solving package specifications: .

UnsatisfiableError: The following specifications were found to be in conflict:
  - caffe-gpu -> cudnn ==5.1
  - cudnn 6.0*
Use "conda info <package>" to see the dependencies for each package.

[jalal@goku crop_align]$
```

Using Python 3, I get this other error. No error on python landmark.py but in the command following it:
```
[jalal@goku crop_align]$ python landmark.py
[jalal@goku crop_align]$ cd ..
[jalal@goku EmotionChallenge]$ ls
cnn  crop_align  data  fact_sheets.pdf  fact_sheets.tex  models  readme.md
[jalal@goku EmotionChallenge]$ cd cnn/
[jalal@goku cnn]$ ls
data_layer.py  deploy.prototxt  extract.py  landmark_224.py  net.prototxt  prepare_data.py  __pycache__  solver.prototxt  train_val.sh  util.py
[jalal@goku cnn]$ python extract.py
Traceback (most recent call last):
  File "extract.py", line 78, in <module>
    submit()
  File "extract.py", line 74, in submit
    extract('../models/final.caffemodel')
  File "extract.py", line 58, in extract
    assert len(landmark) == 136
TypeError: object of type 'map' has no len()
```

If you could help to solve preferrably the one that gives an error for Python 3 and if not the one for Python 2.7. Thanks.

**Comment by monajalal** [comment]
The fix to  `assert len(landmark) == 136` is converting it to   `assert len(list(landmark)) == 136` for Python 3 however I am not aware of the fix for the error I get in Python 2.7 above. I really appreciate if you could share some insight.

**Comment by cleardusk** [comment]
I installed dlib package of Python2.7 using `pip2 install dlib`, you can use conda instead of pip to re-install dlib. The dlib import error indicates that your dlib was not installed rightly.
